Exiled.CustomRoles/API/Features/KeypressActivator.cs

```diff
@@ -0,0 +1,190 @@
+            Log.Warn($"{player.Nickname}: {pressCount} {(player.Role is FpcRole fpc ? fpc.MoveState : false)}");
+            AbilityKeypressTriggerType type = pressCount switch
+            {
+                1 when player.Role is FpcRole { MoveState: PlayerMovementState.Sneaking } => AbilityKeypressTriggerType.DisplayInfo,
+                1 => AbilityKeypressTriggerType.Activate,
+                2 when player.Role is FpcRole { MoveState: PlayerMovementState.Sneaking } => AbilityKeypressTriggerType.SwitchBackward,
+                2 => AbilityKeypressTriggerType.SwitchForward,
+                _ => AbilityKeypressTriggerType.None,
+            };
+
+            player.ShowHint(!PreformAction(player, type, out string response) ? $"Failed to preform action: {response}" : $"Preformed action: {response}", 5f);
+            altTracker[player] = 0;
+        }
+
+        private bool PreformAction(Player player, AbilityKeypressTriggerType type, out string response)
+        {
+            ActiveAbility? selected = player.GetSelectedAbility();
+            if (type == AbilityKeypressTriggerType.Activate)
+            {
+                if (selected is null)
+                {
+                    response = "No selected abilities.";
+                    return false;
+                }
+
+                if (!selected.CanUseAbility(player, out response, CustomRoles.Instance.Config.ActivateOnlySelected))
+                    return false;
+                response = $"{selected.Name} used.";
+                selected.UseAbility(player);
+                return true;
+            }
+
+            if (type is AbilityKeypressTriggerType.SwitchForward or AbilityKeypressTriggerType.SwitchBackward)
+            {
+                List<ActiveAbility> abilities = ListPool<ActiveAbility>.Pool.Get(player.GetActiveAbilities());
+
+                if (abilities.Count == 0)
+                {
+                    response = "No abilities to switch to.";
+                    return false;
+                }
+
+                if (selected is not null)
+                {
+                    int index = abilities.IndexOf(selected);
+                    int mod = type == AbilityKeypressTriggerType.SwitchForward ? 1 : -1;
+                    if (index + mod > abilities.Count - 1)
+                        index = 0;
+                    else if (index + mod < 0)
+                        index = abilities.Count - 1;
+                    else
+                        index += mod;
+
+                    if (index < 0 || index > abilities.Count - 1)
+                    {
+                        Log.Warn("Joker can't do math.");
+                        response = "Jokey did a fucky wucky wif his maths";
+                        return false;
+                    }
+
+                    if (abilities[index] == selected)
+                    {
+                        response = "No abilities to switch to.";
+                        return false;
+                    }
```

Comment:
Why not just checking before the number of ability ? If( .Count <= 1) return false;